### PR TITLE
[Fluent] Changing region to extendable enum

### PR DESCRIPTION
**Pull request by anuchandy**

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@anuchandy__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (anuchan). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

File: src/ResourceManagement/ResourceManager/Microsoft.Azure.Management.ResourceManager.Fluent/Region.cs [hunk 1]
```diff
@@ -1,75 +1,135 @@
...
+
+        private Region(string name)
+        {
+            Name = name.ToLowerInvariant();
+            regions.Add(Name, this);
+        }
+
+        public static Region Create(string name)
+        {
+            name = sWhitespace.Replace(name, "").ToLowerInvariant();
```
Comment (anudeepsharma) [inline src/ResourceManagement/ResourceManager/Microsoft.Azure.Management.ResourceManager.Fluent/Region.cs]: why regex? can't name.replace(' ', '') work?

Comment (anuchandy) [inline src/ResourceManagement/ResourceManager/Microsoft.Azure.Management.ResourceManager.Fluent/Region.cs]: used @"\s+" to take care of all whitespace char types including tab and newline.

File: src/ResourceManagement/ResourceManager/Microsoft.Azure.Management.ResourceManager.Fluent/Region.cs [hunk 2]
```diff
@@ -1,75 +1,135 @@
...
+        public override bool Equals(object obj)
+        {
+
+            string value = this.ToString();
+            if (!(obj is Region))
+            {
+                return false;
+            }
+
+            if (obj == this)
```
Comment (hovsepm) [inline src/ResourceManagement/ResourceManager/Microsoft.Azure.Management.ResourceManager.Fluent/Region.cs]: >if (obj == this) [](start = 12, length = 16)

this comparison produces a valid warning "Possible unintended reference comparison; to get a value comparison, cast the left hand side to type". Please fix it here and in all other cases that you've already merged.

**Pull request merged**